pkg/restore/util.go

```diff
@@ -71,32 +77,48 @@ func GetRewriteRules(
 	oldTable *model.TableInfo,
 	newTimeStamp uint64,
 ) *restore_util.RewriteRules {
-	tableRules := make([]*import_sstpb.RewriteRule, 0, 1)
-	tableRules = append(tableRules, &import_sstpb.RewriteRule{
-		OldKeyPrefix: tablecodec.EncodeTablePrefix(oldTable.ID),
-		NewKeyPrefix: tablecodec.EncodeTablePrefix(newTable.ID),
-		NewTimestamp: newTimeStamp,
-	})
-
-	dataRules := make([]*import_sstpb.RewriteRule, 0, len(oldTable.Indices)+1)
-	dataRules = append(dataRules, &import_sstpb.RewriteRule{
-		OldKeyPrefix: append(tablecodec.EncodeTablePrefix(oldTable.ID), recordPrefixSep...),
-		NewKeyPrefix: append(tablecodec.EncodeTablePrefix(newTable.ID), recordPrefixSep...),
-		NewTimestamp: newTimeStamp,
-	})
-
+	tableIDs := make(map[int64]int64)
+	tableIDs[oldTable.ID] = newTable.ID
```

Comment:
if we got `oldTable.Partition != nil` then tableID is useless
should it be?
```
if oldTable.Partition != nil {
	for _, srcPart := range oldTable.Partition.Definitions {
		for _, destPart := range newTable.Partition.Definitions {
			if srcPart.Name == destPart.Name {
				tableIDs[srcPart.ID] = destPart.ID
			}
		}
	}
} else {
	tableIDs[oldTable.ID] = newTable.ID
}
```

Reply:
We don't have partitioned index so tableID is necessary.
BTW I think extra rewrite rules do not have a bad effect.

Reply:
yes, extra rewrite rules seems no bad effect